src/lib/storage/reference_column/reference_column_iterable.hpp

```diff
@@ -17,74 +15,96 @@ class ReferenceColumnIterable : public ColumnIterable<ReferenceColumnIterable<T>
 
   template <typename Functor>
   void _on_with_iterators(const Functor& functor) const {
-    const auto table = _column.referenced_table();
-    const auto column_id = _column.referenced_column_id();
+    const auto pos_list = _column.pos_list();
+
+    const auto chunk_id_and_chunk_offsets_list = to_chunk_id_and_chunk_offsets_list(*pos_list);
```

Comment:
Could you add a test for this new feature? Also, do the existing tests cover the case where the chunk id is not unique?

Reply:
You mean a test for this function? The tests should cover this case because it happens whenever a reference column created by a join operator is accessed.

Reply:
for everything. I wouldn't rely on unrelated tests to cover this. We now have a shortcut and relevant tests should make sure to cover both the fast and the slow path.